src/Main.elm

```diff
@@ -143,18 +143,43 @@ loginScreen =
         ]
 
 
-mainScreen : User.User -> H.Html Msg
-mainScreen user =
-    H.div []
-        [ mainScreenHeader user
-        , H.main_ []
-            [ H.div [ HA.class "mdc-layout-grid" ]
-                [ H.div [ HA.class "mdc-layout-grid__inner" ]
-                    [ H.div [ HA.class "mdc-layout-grid__cell" ] []
+mainScreen : AuthedPageState -> H.Html Msg
+mainScreen { user, threads, messageIds } =
+    let
+        allData =
+            RemoteData.map2 (,) threads messageIds
+                |> RemoteData.map
+                    (\( threads, messageIds ) -> ( threads, messageIds ))
+
+        threadView messages thread =
+            H.div []
+                [ H.h3 [ HA.class "mdc-list-group__subheader" ] [ H.text thread.snippet ]
+                , H.ul [ HA.class "mdc-list" ]
+                    [ List.filter (\message -> message.threadId == thread.threadId) messages
+                        |> List.map (\message -> H.li [ HA.class "mdc-list-item" ] [ H.text <| toString message.messageId ])
+                        |> H.ul [ HA.class "mdc-list" ]
                     ]
                 ]
+    in
+        H.div []
+            [ mainScreenHeader user
+            , H.main_ []
+                [ C.progressBar allData
+                , case allData of
```

Comment:
It doesn't look like you are using `allData` anywhere else so you can just inline `case RemoteData.map2 (,) threads messageIds of`.

Reply:
Actually, I will replace that part (or rather remodel that). Messages are grouped within threads in GMail, so makes sense only to load the threads and inside the `ThreadView` load all corresponding messages.